src/garage/tf/algos/rl2.py

```diff
@@ -185,9 +332,18 @@ def _process_samples(self, itr, paths):
             np_tensor_utils.stack_and_pad_tensor_dict_list(
                 concatenated_paths, self._inner_algo.max_path_length))
 
-        undiscounted_returns = log_performance(
-            itr, TrajectoryBatch.from_trajectory_list(self._env_spec, paths),
-            self._inner_algo.discount)
+        name_map = None
+        if hasattr(self._task_sampler._envs[0].env, 'all_task_names'):
+            names = [
+                env.env.all_task_names[0] for env in self._task_sampler._envs
+            ]
+            name_map = dict(zip(names, names))
```

Comment:
`name_map` does not change over the training. Can it can precomputed in `__init__()`?

Reply:
This also isn't actually the correct type for `name_map`. It should map `int -> str` (i.e. `dict(enumerate(names))`).